torch/lib/c10d/CMakeLists.txt

```diff
@@ -7,32 +7,29 @@ list(APPEND CMAKE_MODULE_PATH
   ${CMAKE_CURRENT_SOURCE_DIR}/../../../cmake/Modules_CUDA_fix)
 
 if(USE_CUDA)
-  find_package(CUDA)
-  if(CUDA_FOUND)
-    set(C10D_USE_CUDA true)
-    message(STATUS "Building C10D with CUDA support")
-    add_definitions(-DUSE_CUDA=1)
-  else()
-    set(C10D_USE_CUDA false)
-    message(STATUS "CUDA not found, building C10D without CUDA support")
-  endif()
+  add_definitions(-DUSE_CUDA=1)
 elseif(USE_ROCM)
   add_definitions(-DUSE_ROCM=1)
   add_definitions(-D__HIP_PLATFORM_HCC__=1)
   add_definitions(-DHIP_VERSION=${HIP_VERSION_MAJOR})
-
-  set(C10D_USE_CUDA false)
-  set(C10D_USE_ROCM true)
 else()
-  set(C10D_USE_CUDA false)
-  message(STATUS "Building C10D without CUDA support")
+  message(STATUS "Building c10d without CUDA/ROCm support")
+endif()
+
+if(USE_GLOO)
+  option(USE_C10D_GLOO "USE C10D GLOO" ON)
+endif()
+
+if(USE_NCCL)
```

Comment:
what if `USE_NCCL` is ON but `USE_CUDA` is off?

Reply:
This case is handled earlier by `cmake/Dependencies.cmake` and `USE_NCCL` is set to `OFF`.